From:   Hou Tao <houtao1@huawei.com>
To:     <linux-xfs@vger.kernel.org>
CC:     <darrick.wong@oracle.com>, <cmaiolino@redhat.com>,
        <sandeen@redhat.com>
Subject: [PATCH] xfs: reset b_first_retry_time when clear the retry status of xfs_buf_t
Date:   Fri, 3 Feb 2017 17:08:25 +0800

After successful IO or permanent error, b_first_retry_time also
needs to be cleared, else the invalid first retry time will be
used by the next retry check.

Signed-off-by: Hou Tao <houtao1@huawei.com>
---
 fs/xfs/xfs_buf_item.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/xfs/xfs_buf_item.c b/fs/xfs/xfs_buf_item.c
index 2975cb2..0306168 100644
--- a/fs/xfs/xfs_buf_item.c
+++ b/fs/xfs/xfs_buf_item.c
@@ -1162,6 +1162,7 @@ xfs_buf_iodone_callbacks(
 	 */
 	bp->b_last_error = 0;
 	bp->b_retries = 0;
+	bp->b_first_retry_time = 0;
 
 	xfs_buf_do_callbacks(bp);
 	bp->b_fspriv = NULL;
